src/cascadia/ut_app/DynamicProfileTests.cpp

```diff
@@ -0,0 +1,677 @@
+        // then that of a profile without a source.
+
+        auto gen0 = std::make_unique<TestDynamicProfileGenerator>(L"Terminal.App.UnitTest.0");
+        gen0->pfnGenerate = []() {
+            std::vector<Profile> profiles;
+            Profile p0;
+            p0.SetName(L"profile0"); // this is _profiles.at(2)
+            profiles.push_back(p0);
+            return profiles;
+        };
+        auto gen1 = std::make_unique<TestDynamicProfileGenerator>(L"Terminal.App.UnitTest.1");
+        gen1->pfnGenerate = []() {
+            std::vector<Profile> profiles;
+            Profile p0, p1;
+            p0.SetName(L"profile0"); // this is _profiles.at(3)
+            p1.SetName(L"profile1"); // this is _profiles.at(4)
+            profiles.push_back(p0);
+            profiles.push_back(p1);
+            return profiles;
+        };
+
+        CascadiaSettings settings{ false };
+        settings._profileGenerators.emplace_back(std::move(gen0));
+        settings._profileGenerators.emplace_back(std::move(gen1));
+
+        Profile p0, p1;
+        p0.SetName(L"profile0"); // this is _profiles.at(0)
+        p1.SetName(L"profile1"); // this is _profiles.at(1)
+        settings._profiles.push_back(p0);
+        settings._profiles.push_back(p1);
+
+        settings._LoadDynamicProfiles();
+        VERIFY_ARE_EQUAL(5u, settings._profiles.size());
+
+        VERIFY_ARE_EQUAL(L"profile0", settings._profiles.at(0).GetName());
+        VERIFY_IS_FALSE(settings._profiles.at(0)._guid.has_value());
+        VERIFY_IS_FALSE(settings._profiles.at(0)._source.has_value());
+
+        VERIFY_ARE_EQUAL(L"profile1", settings._profiles.at(1).GetName());
+        VERIFY_IS_FALSE(settings._profiles.at(1)._guid.has_value());
+        VERIFY_IS_FALSE(settings._profiles.at(1)._source.has_value());
+
+        VERIFY_ARE_EQUAL(L"profile0", settings._profiles.at(2).GetName());
+        VERIFY_IS_FALSE(settings._profiles.at(2)._guid.has_value());
+        VERIFY_IS_TRUE(settings._profiles.at(2)._source.has_value());
+
+        VERIFY_ARE_EQUAL(L"profile0", settings._profiles.at(3).GetName());
+        VERIFY_IS_FALSE(settings._profiles.at(3)._guid.has_value());
+        VERIFY_IS_TRUE(settings._profiles.at(3)._source.has_value());
+
+        VERIFY_ARE_EQUAL(L"profile1", settings._profiles.at(4).GetName());
+        VERIFY_IS_FALSE(settings._profiles.at(4)._guid.has_value());
+        VERIFY_IS_TRUE(settings._profiles.at(4)._source.has_value());
+
+        settings._ValidateProfilesHaveGuid();
+
+        VERIFY_IS_TRUE(settings._profiles.at(0)._guid.has_value());
+        VERIFY_IS_TRUE(settings._profiles.at(1)._guid.has_value());
+        VERIFY_IS_TRUE(settings._profiles.at(2)._guid.has_value());
+        VERIFY_IS_TRUE(settings._profiles.at(3)._guid.has_value());
+        VERIFY_IS_TRUE(settings._profiles.at(4)._guid.has_value());
+
+        VERIFY_ARE_NOT_EQUAL(settings._profiles.at(0)._guid.value(),

```

Comment:
Here we're verifying that all of the GUIDs are unique right? Why not loop through every possible combination to ensure that? (or, better yet, push all of the guids into a set and check if the GUID actually got inserted)